runtime/src/main/scala/com/ing/baker/runtime/core/interations/InteractionManager.scala

```diff
@@ -2,21 +2,49 @@ package com.ing.baker.runtime.core.interations
 
 import com.ing.baker.il.petrinet.InteractionTransition
 
+/**
+  * The InteractionManager is responsible for all implementation of interactions.
+  * It knows all available implementations and gives the correct implementation for an Interaction
+  * @param interactionImplementations All
+  */
 class InteractionManager(private var interactionImplementations: Seq[InteractionImplementation] = Seq.empty) {
 
-  def add(interactionImplementation: InteractionImplementation) =
-    interactionImplementations :+= interactionImplementation
+  /**
+    * Add an implementation to the InteractionManager
+    * @param implementation
+    */
+  def addImplementation(implementation: InteractionImplementation) =
+    interactionImplementations :+= implementation
 
-  def remove(interactionImplementation: InteractionImplementation) =
-    interactionImplementations = interactionImplementations.filter(_.name != interactionImplementation.name)
+  /**
+    * Gets an implementation is available for the given interaction.
+    * It checks:
+    *   1. Name
+    *   2. Input variable sizes
+    *   3. Input variable types
+    * @param interaction The interaction to check
+    * @return An option containing the implementation if available
+    */
+  def getImplementation(interaction: InteractionTransition[_]) : Option[InteractionImplementation] =
+    interactionImplementations.find(implementation => isCompatibleImplementation(interaction, implementation))
```

Comment:
It would be nice if this `isCompatibleImplementation` outcome could be cached since it is invoked for each execution of an interaction.